src/item_factory.cpp

```diff
@@ -510,14 +516,16 @@ void Item_factory::finalize_post( itype &obj )
     if( obj.armor ) {
         // Setting max_encumber must be in finalize_post because it relies on
         // stack_size being set for all ammo, which happens in finalize_pre.
-        if( obj.armor->max_encumber < 0 ) {
-            units::volume total_nonrigid_volume = 0_ml;
-            for( const pocket_data &pocket : obj.pockets ) {
-                if( !pocket.rigid ) {
-                    total_nonrigid_volume += pocket.max_contains_volume();
+        for( random_armor_data &data : obj.armor->data ) {
+            if( data.max_encumber == 0 ) {
+                units::volume total_nonrigid_volume = 0_ml;
+                for( const pocket_data &pocket : obj.pockets ) {
+                    if( !pocket.rigid ) {
+                        total_nonrigid_volume += pocket.max_contains_volume();
+                    }
                 }
+                data.max_encumber = data.encumber + total_nonrigid_volume / 250_ml;
             }
```

Comment:
I feel that this should remain checking for negative `max_encumber` (and that `max_encumber` should default to -1), because you might want `encumber` and `max_encumber` to both be zero, and not invoke this fallback code.

Reply:
I believe it always checked for 0. but I agree, -1 is a much more sensible default value.